### ISSUE TITLE
Analysis of Results

### ISSUE THREAD

**Issue by scottfernandes** [jacobf18/tabular#5]
Hi, I would like to analyze the results after imputation, is there anyway I could see the probabilities as tabular models use conditional probability under the hood,  or any other way for analysis

**Comment by jacobf18** [comment]
@scottfernandes That functionality isn't there at the moment but it wouldn't be hard to add. You would just need to pull out the `preds` variable in the following code (if you are using TabImpute v2 interface):

```python
def _get_imputation_single(
        self, X_normalized: np.ndarray, num_repeats: int = 1
    ) -> np.ndarray:
        if num_repeats > 1:
            raise ValueError("`num_repeats > 1` is not supported in TabImputeV2.")

        X_normalized_tensor = torch.from_numpy(X_normalized).to(self.device)
        with torch.no_grad():
            X_normalized_tensor = X_normalized_tensor.unsqueeze(0).to(torch.bfloat16)
            preds = self.model(X_normalized_tensor)
            medians = self.bar_distribution.median(logits=preds).squeeze(0)

        X_imputed = medians.cpu().detach().numpy()
        missing_mask = np.isnan(X_normalized)
        X_normalized[missing_mask] = X_imputed[missing_mask]
        return X_normalized, X_imputed
```

The preds variable is the logits and will be shaped (1, num_rows, num_cols, 5000) where the 5000 is over the binned space. You would then use the `self.borders` FulllBarDistribution (check out the tabimpute.model.bar_distribution file) which would let you get any relevant quantities you would want (quantiles, variance, mean, etc.). However, before using those values directly, note that the matrix is normalized per column by subtracting mean and dividing by standard deviation, so the values used in the `self.borders` file would need to be "denormalized" back to the original space.
